src/providers/ipa/ipa_deskprofile_rules.c

```diff
@@ -0,0 +1,343 @@
+                                  IPA_MEMBER_HOST, host_dn_clean);
+    if (rule_filter == NULL) {
+        ret = ENOMEM;
+        goto immediate;
+    }
+
+    /* Add all parent groups of ipa_hostname to the filter */
+    ret = sysdb_attrs_get_string_array(ipa_host, SYSDB_ORIG_MEMBEROF,
+                                       tmp_ctx, &memberof_list);
+    if (ret != EOK && ret != ENOENT) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Could not identify.\n");
+    } if (ret == ENOENT) {
+        /* This host is not a member of any hostgroups */
+        memberof_list = talloc_array(tmp_ctx, const char *, 1);
+        if (memberof_list == NULL) {
+            ret = ENOMEM;
+            goto immediate;
+        }
+        memberof_list[0] = NULL;
+    }
+
+    for (size_t i = 0; memberof_list[i]; i++) {
+        ret = sss_filter_sanitize(tmp_ctx,
+                                  memberof_list[i],
+                                  &host_group_clean);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_CRIT_FAILURE,
+                  "sss_filter_sanitize() failed [%d]: %s\n",
+                  ret, sss_strerror(ret));
+            goto immediate;
+        }
+
+        rule_filter = talloc_asprintf_append(rule_filter, "(%s=%s)",
+                                             IPA_MEMBER_HOST,
+                                             host_group_clean);
+        if (rule_filter == NULL) {
+            ret = ENOMEM;
+            goto immediate;
+        }
+    }
+
+    ret = get_sysdb_grouplist(tmp_ctx, domain->sysdb, domain, username,
+                              &groups_list);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "get_sysdb_grouplist() failed [%d]: %s\n",
+              ret, sss_strerror(ret));
+        goto immediate;
+    }
+
+    for (size_t i = 0; groups_list[i] != NULL; i++) {
+        ret = sss_filter_sanitize(tmp_ctx, groups_list[i], &group_clean);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_CRIT_FAILURE,
+                  "sss_filter_sanitize() failed [%d]: %s\n",
+                  ret, sss_strerror(ret));
+            goto immediate;
+        }
+
+        ret  = sss_parse_internal_fqname(tmp_ctx, group_clean, &group, NULL);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_CRIT_FAILURE,
+                  "sss_parse_internal_fqname() failed [%d]: %s\n",
+                  ret, sss_strerror(ret));
+            goto immediate;
+        }
+
+        rule_filter = talloc_asprintf_append(rule_filter, "(%s=%s)",
+                                             IPA_MEMBER_USER, group);
+        if (rule_filter == NULL) {
+            ret = ENOMEM;
+            goto immediate;
+        }
+    }
+
+    rule_filter = talloc_asprintf_append(rule_filter, "))");
+    if (rule_filter == NULL) {
+        ret = ENOMEM;
+        goto immediate;
+    }
+    state->rules_filter = talloc_steal(state, rule_filter);
+
+    ret = ipa_deskprofile_rule_info_next(req, state);
+    if (ret == EOK) {
+        ret = EINVAL;
+    }
+
```

Comment:
So, basically it happens because _rule_info_next() expect a base during the first iteration, but also returns EOK in case the base is NULL (as it's okay to happen in the subsequent iterations).

I've added a comment in order to make it a little bit more clear in both deskprofile and hbac code.

So, fixed locally!